Realm/ObjectStore/handover.hpp

```diff
@@ -0,0 +1,120 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef handover_hpp
+#define handover_hpp
+
+#include <realm/row.hpp>
+#include <realm/query.hpp>
+#include <realm/group_shared.hpp>
+#include <realm/link_view.hpp>
+#include <realm/table_view.hpp>
+
+namespace realm {
+    class HandoverPackage;
+    class SharedGroup;
+
+    class AnyHandover;
+
+    // Type-erased wrapper for any type which must be exported to be handed between threads
+    class AnyThreadConfined {
+    public:
```

Comment:
Our usual class member ordering is all public, then all private, with things within the sections being data members, constructors/etc., then other member functions.